beacon-chain/rpc/eth/light-client/helpers_test.go

```diff
@@ -0,0 +1,439 @@
+					SyncCommitteeBits: []byte{0b00111100, 0b1}, // [0,0,1,1,1,1,0,0]
+				},
+				AttestedHeader: &ethpbv1.BeaconBlockHeader{
+					Slot: 1000001,
+				},
+				NextSyncCommitteeBranch: make([][]byte, 4),
+				SignatureSlot:           1000000,
+			},
+			expectedResult: true,
+		},
+		{
+			name: "old has relevant sync committee but new doesn't",
+			oldUpdate: &ethpbv2.LightClientUpdate{
+				SyncAggregate: &ethpbv1.SyncAggregate{
+					SyncCommitteeBits: []byte{0b00111100, 0b1}, // [0,0,1,1,1,1,0,0]
+				},
+				AttestedHeader: &ethpbv1.BeaconBlockHeader{
+					Slot: 1000001,
+				},
+				NextSyncCommitteeBranch: make([][]byte, 4),
+				SignatureSlot:           1000000,
+			},
+			newUpdate: &ethpbv2.LightClientUpdate{
+				SyncAggregate: &ethpbv1.SyncAggregate{
+					SyncCommitteeBits: []byte{0b00111100, 0b1}, // [0,0,1,1,1,1,0,0]
+				},
+				AttestedHeader: &ethpbv1.BeaconBlockHeader{
+					Slot: 1000000,
+				},
+				NextSyncCommitteeBranch: make([][]byte, fieldparams.NextSyncCommitteeBranchDepth),
+				SignatureSlot:           9999,
+			},
+			expectedResult: false,
+		},
+		{
+			name: "new has finality but old doesn't",
+			oldUpdate: &ethpbv2.LightClientUpdate{
+				SyncAggregate: &ethpbv1.SyncAggregate{
+					SyncCommitteeBits: []byte{0b00111100, 0b1}, // [0,0,1,1,1,1,0,0]
+				},
+				AttestedHeader: &ethpbv1.BeaconBlockHeader{
+					Slot: 1000000,
+				},
+				NextSyncCommitteeBranch: make([][]byte, 4),
+				SignatureSlot:           9999,
+				FinalityBranch:          make([][]byte, blockchain.FinalityBranchNumOfLeaves),
+			},
+			newUpdate: &ethpbv2.LightClientUpdate{
+				SyncAggregate: &ethpbv1.SyncAggregate{
+					SyncCommitteeBits: []byte{0b00111100, 0b1}, // [0,0,1,1,1,1,0,0]
+				},
+				AttestedHeader: &ethpbv1.BeaconBlockHeader{
+					Slot: 1000000,
+				},
+				NextSyncCommitteeBranch: make([][]byte, 4),
+				SignatureSlot:           9999,
+				FinalityBranch:          make([][]byte, 5),
+			},
+			expectedResult: true,
+		},
+		{
+			name: "old has finality but new doesn't",
+			oldUpdate: &ethpbv2.LightClientUpdate{
+				SyncAggregate: &ethpbv1.SyncAggregate{
+					SyncCommitteeBits: []byte{0b00111100, 0b1}, // [0,0,1,1,1,1,0,0]
+				},
+				AttestedHeader: &ethpbv1.BeaconBlockHeader{
+					Slot: 1000000,
+				},
+				NextSyncCommitteeBranch: make([][]byte, 4),
+				SignatureSlot:           9999,
+				FinalityBranch:          make([][]byte, 5),
+			},
+			newUpdate: &ethpbv2.LightClientUpdate{
+				SyncAggregate: &ethpbv1.SyncAggregate{
+					SyncCommitteeBits: []byte{0b00111100, 0b1}, // [0,0,1,1,1,1,0,0]
+				},
+				AttestedHeader: &ethpbv1.BeaconBlockHeader{
+					Slot: 1000000,
+				},
+				NextSyncCommitteeBranch: make([][]byte, 4),
+				SignatureSlot:           9999,
+				FinalityBranch:          make([][]byte, blockchain.FinalityBranchNumOfLeaves),
+			},
+			expectedResult: false,
+		},
+		{
+			name: "new has finality and sync committee finality both but old doesn't have sync committee finality",
```

Comment:
Although the test itself is good (either both need to have finality or both need to not have finality), the name says "new has finality" but you set the number of items of `FinalityBranch` to `5` instead of `FinalityBranchNumOfLeaves`

Reply:
according to https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/sync-protocol.md#is_finality_update,  new has finality means that number of items of ``FinalityBranch`` should not be equal to ``FinalityBranchNumOfLeaves`` (i.e. 6) right? Or am I missing something?